test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -383,8 +391,62 @@ void ConfigureTestServices(IServiceCollection services)
             Assert.Single(activityProcessor.Invocations, invo => invo.Method.Name == "OnEnd");
             var activity = activityProcessor.Invocations.FirstOrDefault(invo => invo.Method.Name == "OnEnd").Arguments[0] as Activity;
 
-            Assert.Equal(ActivityKind.Server, activity.Kind);
-            Assert.Equal("/api/values", activity.GetTagValue(SpanAttributeConstants.HttpPathKey) as string);
+            ValidateAspNetCoreActivity(activity, "/api/values");
+        }
+
+        [Fact]
+        public async Task PropagateContextIrrespectiveOfSamplingDecision()
+        {
+            var activityProcessor = new Mock<BaseProcessor<Activity>>();
+
+            var expectedTraceId = ActivityTraceId.CreateRandom();
+            var expectedParentSpanId = ActivitySpanId.CreateRandom();
+
+            // Arrange
+            using (var testFactory = this.factory
+                .WithWebHostBuilder(builder =>
+                    builder.ConfigureTestServices(services =>
+                    {
+                        this.openTelemetrySdk = Sdk.CreateTracerProviderBuilder()
+                        .SetSampler(new AlwaysOffSampler())
+                        .AddAspNetCoreInstrumentation()
+                        .AddProcessor(activityProcessor.Object)
+                        .Build();
+                    })))
+            {
+                using var client = testFactory.CreateClient();
+
+                // Test TraceContext Propagation
+                var expectedTraceState = "rojo=1,congo=2";
+                var request = new HttpRequestMessage(HttpMethod.Get, "/api/GetChildActivityTraceContext");
+                request.Headers.Add("traceparent", $"00-{expectedTraceId}-{expectedParentSpanId}-01");
+                request.Headers.Add("tracestate", expectedTraceState);
+
+                var response = await client.SendAsync(request);
+                var childActivityTraceContext = JsonConvert.DeserializeObject<Dictionary<string, string>>(response.Content.ReadAsStringAsync().Result);
+
+                response.EnsureSuccessStatusCode();
+
+                Assert.Equal(expectedTraceId.ToString(), childActivityTraceContext["TraceId"]);
+                Assert.Equal(expectedTraceState, childActivityTraceContext["TraceState"]);
+                Assert.NotEqual(expectedParentSpanId.ToString(), childActivityTraceContext["ParentSpanId"]); // there is a new activity created in instrumentation therefore the ParentSpanId is different that what is provided in the headers
+
+                // Test Baggage Context Propagation
+                request = new HttpRequestMessage(HttpMethod.Get, "/api/GetChildActivityBaggageContext");
+                request.Headers.Add("Baggage", "key1=value1,key2=value2");
+                response = await client.SendAsync(request);
+                var childActivityBaggageContext = JsonConvert.DeserializeObject<IReadOnlyDictionary<string, string>>(response.Content.ReadAsStringAsync().Result);
```

Comment:
you can avoid the additional json parsing, by making the test app return 2 strings contaning traceparent and tracestate (activity.Id, activity.TraceState) respectively.